File: task_equivalents/llm_translations/starcoder/optimized/python/data_visualization/01_scatter_trend_grouped.py
import pandas as pd
auto = pd.read_csv('data/csv/Auto.csv', na_values='?')
auto = auto.dropna()
n = len(auto)
X = auto['horsepower'].to_numpy().reshape((-1, 1))
Y = auto['mpg'].to_numpy().reshape((-1, 1))
X = np.column_stack((np.ones(n), X))
model1 = LinearRegression().fit(X[:, 1].reshape((-1, 1)), Y)
a = model1.intercept_[0]
b = model1.coef_[0][0]
yfit = a + b * X[:, 1]
plt.scatter(X[:, 1], Y, label='Data')
plt.plot(X[:, 1], yfit, color='red', label='Regression Line')
plt.xlabel('horsepower')
plt.ylabel('mpg')
plt.legend()
plt.show()

File: task_equivalents/llm_translations/starcoder/optimized/r/data_visualization/01_scatter_trend_grouped.R
library(readr)
library(dplyr)
library(broom)
library(ggplot2)
Auto <- read_csv('data/csv/Auto.csv', na = '?') %>% drop_na()
n  = nrow(Auto)
X  = matrix(Auto$horsepower, ncol = 1)
Y  = matrix(Auto$mpg, ncol = 1)
X  = cbind(rep(1, n), X)
model1_fit  = lm(Y ~ X[, 2])
a  = coef(model1_fit)[1]
b  = coef(model1_fit)[2]
yfit  = a + b * X[, 2]
ggplot(Auto, aes(x = horsepower, y = mpg)) +
  geom_point() +
  geom_line(aes(y = yfit), color = 'red') +
  xlab('horsepower') +
  ylab('mpg')
